// @hathora/hathora-cloud-sdk/dist/esm/models/AchPaymentMethod.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the AchPaymentMethod interface.
 */
export function instanceOfAchPaymentMethod(value) {
    let isInstance = true;
    return isInstance;
}
export function AchPaymentMethodFromJSON(json) {
    return AchPaymentMethodFromJSONTyped(json, false);
}
export function AchPaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'last4': !exists(json, 'last4') ? undefined : json['last4'],
        'bankName': !exists(json, 'bankName') ? undefined : json['bankName'],
    };
}
export function AchPaymentMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'last4': value.last4,
        'bankName': value.bankName,
    };
}
