// @hathora/hathora-cloud-sdk/dist/esm/models/AppConfigAuthConfiguration.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ApplicationAuthConfigurationGoogleFromJSON, ApplicationAuthConfigurationGoogleToJSON, } from './ApplicationAuthConfigurationGoogle';
/**
 * Check if a given object implements the AppConfigAuthConfiguration interface.
 */
export function instanceOfAppConfigAuthConfiguration(value) {
    let isInstance = true;
    return isInstance;
}
export function AppConfigAuthConfigurationFromJSON(json) {
    return AppConfigAuthConfigurationFromJSONTyped(json, false);
}
export function AppConfigAuthConfigurationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !exists(json, 'google') ? undefined : ApplicationAuthConfigurationGoogleFromJSON(json['google']),
        'nickname': !exists(json, 'nickname') ? undefined : json['nickname'],
        'anonymous': !exists(json, 'anonymous') ? undefined : json['anonymous'],
    };
}
export function AppConfigAuthConfigurationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': ApplicationAuthConfigurationGoogleToJSON(value.google),
        'nickname': value.nickname,
        'anonymous': value.anonymous,
    };
}
