// @hathora/hathora-cloud-sdk/dist/esm/models/ApplicationWithDeploymentAllOf.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { DeploymentFromJSON, DeploymentToJSON, } from './Deployment';
/**
 * Check if a given object implements the ApplicationWithDeploymentAllOf interface.
 */
export function instanceOfApplicationWithDeploymentAllOf(value) {
    let isInstance = true;
    isInstance = isInstance && "deployment" in value;
    return isInstance;
}
export function ApplicationWithDeploymentAllOfFromJSON(json) {
    return ApplicationWithDeploymentAllOfFromJSONTyped(json, false);
}
export function ApplicationWithDeploymentAllOfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deployment': DeploymentFromJSON(json['deployment']),
    };
}
export function ApplicationWithDeploymentAllOfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'deployment': DeploymentToJSON(value.deployment),
    };
}
