// @hathora/hathora-cloud-sdk/dist/esm/models/ConnectionInfoV2.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ExposedPortFromJSON, ExposedPortToJSON, } from './ExposedPort';
/**
 * @export
 */
export const ConnectionInfoV2StatusEnum = {
    Starting: 'starting',
    Active: 'active'
};
/**
 * Check if a given object implements the ConnectionInfoV2 interface.
 */
export function instanceOfConnectionInfoV2(value) {
    let isInstance = true;
    isInstance = isInstance && "additionalExposedPorts" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "roomId" in value;
    return isInstance;
}
export function ConnectionInfoV2FromJSON(json) {
    return ConnectionInfoV2FromJSONTyped(json, false);
}
export function ConnectionInfoV2FromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'additionalExposedPorts': (json['additionalExposedPorts'].map(ExposedPortFromJSON)),
        'exposedPort': !exists(json, 'exposedPort') ? undefined : ExposedPortFromJSON(json['exposedPort']),
        'status': json['status'],
        'roomId': json['roomId'],
    };
}
export function ConnectionInfoV2ToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'additionalExposedPorts': (value.additionalExposedPorts.map(ExposedPortToJSON)),
        'exposedPort': ExposedPortToJSON(value.exposedPort),
        'status': value.status,
        'roomId': value.roomId,
    };
}
