// @hathora/hathora-cloud-sdk/dist/esm/models/ContainerPort.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { TransportTypeFromJSON, TransportTypeToJSON, } from './TransportType';
/**
 * Check if a given object implements the ContainerPort interface.
 */
export function instanceOfContainerPort(value) {
    let isInstance = true;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function ContainerPortFromJSON(json) {
    return ContainerPortFromJSONTyped(json, false);
}
export function ContainerPortFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'transportType': TransportTypeFromJSON(json['transportType']),
        'port': json['port'],
        'name': json['name'],
    };
}
export function ContainerPortToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transportType': TransportTypeToJSON(value.transportType),
        'port': value.port,
        'name': value.name,
    };
}
