// @hathora/hathora-cloud-sdk/dist/esm/models/CreateLobbyRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { LobbyVisibilityFromJSON, LobbyVisibilityToJSON, } from './LobbyVisibility';
import { RegionFromJSON, RegionToJSON, } from './Region';
/**
 * Check if a given object implements the CreateLobbyRequest interface.
 */
export function instanceOfCreateLobbyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "visibility" in value;
    isInstance = isInstance && "initialConfig" in value;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
export function CreateLobbyRequestFromJSON(json) {
    return CreateLobbyRequestFromJSONTyped(json, false);
}
export function CreateLobbyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'visibility': LobbyVisibilityFromJSON(json['visibility']),
        'initialConfig': json['initialConfig'],
        'region': RegionFromJSON(json['region']),
    };
}
export function CreateLobbyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'visibility': LobbyVisibilityToJSON(value.visibility),
        'initialConfig': value.initialConfig,
        'region': RegionToJSON(value.region),
    };
}
