// @hathora/hathora-cloud-sdk/dist/esm/models/CreateRoomRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { RegionFromJSON, RegionToJSON, } from './Region';
/**
 * Check if a given object implements the CreateRoomRequest interface.
 */
export function instanceOfCreateRoomRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
export function CreateRoomRequestFromJSON(json) {
    return CreateRoomRequestFromJSONTyped(json, false);
}
export function CreateRoomRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'region': RegionFromJSON(json['region']),
    };
}
export function CreateRoomRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'region': RegionToJSON(value.region),
    };
}
