// @hathora/hathora-cloud-sdk/dist/esm/models/CreateRoomDeprecatedRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { RegionFromJSON, RegionToJSON, } from './Region';
/**
 * Check if a given object implements the CreateRoomDeprecatedRequest interface.
 */
export function instanceOfCreateRoomDeprecatedRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
export function CreateRoomDeprecatedRequestFromJSON(json) {
    return CreateRoomDeprecatedRequestFromJSONTyped(json, false);
}
export function CreateRoomDeprecatedRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'region': RegionFromJSON(json['region']),
    };
}
export function CreateRoomDeprecatedRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'region': RegionToJSON(value.region),
    };
}
