// @hathora/hathora-cloud-sdk/dist/esm/models/CustomerPortalUrl.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the CustomerPortalUrl interface.
 */
export function instanceOfCustomerPortalUrl(value) {
    let isInstance = true;
    isInstance = isInstance && "returnUrl" in value;
    return isInstance;
}
export function CustomerPortalUrlFromJSON(json) {
    return CustomerPortalUrlFromJSONTyped(json, false);
}
export function CustomerPortalUrlFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'returnUrl': json['returnUrl'],
    };
}
export function CustomerPortalUrlToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'returnUrl': value.returnUrl,
    };
}
