// @hathora/hathora-cloud-sdk/dist/esm/models/DeploymentConfig.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ContainerPortFromJSON, ContainerPortToJSON, } from './ContainerPort';
import { DeploymentEnvInnerFromJSON, DeploymentEnvInnerToJSON, } from './DeploymentEnvInner';
import { PlanNameFromJSON, PlanNameToJSON, } from './PlanName';
import { TransportTypeFromJSON, TransportTypeToJSON, } from './TransportType';
/**
 * Check if a given object implements the DeploymentConfig interface.
 */
export function instanceOfDeploymentConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "env" in value;
    isInstance = isInstance && "roomsPerProcess" in value;
    isInstance = isInstance && "planName" in value;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "containerPort" in value;
    return isInstance;
}
export function DeploymentConfigFromJSON(json) {
    return DeploymentConfigFromJSONTyped(json, false);
}
export function DeploymentConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'env': (json['env'].map(DeploymentEnvInnerFromJSON)),
        'roomsPerProcess': json['roomsPerProcess'],
        'planName': PlanNameFromJSON(json['planName']),
        'additionalContainerPorts': !exists(json, 'additionalContainerPorts') ? undefined : (json['additionalContainerPorts'].map(ContainerPortFromJSON)),
        'transportType': TransportTypeFromJSON(json['transportType']),
        'containerPort': json['containerPort'],
    };
}
export function DeploymentConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'env': (value.env.map(DeploymentEnvInnerToJSON)),
        'roomsPerProcess': value.roomsPerProcess,
        'planName': PlanNameToJSON(value.planName),
        'additionalContainerPorts': value.additionalContainerPorts === undefined ? undefined : (value.additionalContainerPorts.map(ContainerPortToJSON)),
        'transportType': TransportTypeToJSON(value.transportType),
        'containerPort': value.containerPort,
    };
}
