// @hathora/hathora-cloud-sdk/dist/esm/models/Invoice.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * @export
 */
export const InvoiceStatusEnum = {
    Pending: 'pending',
    Paid: 'paid',
    Overdue: 'overdue'
};
/**
 * Check if a given object implements the Invoice interface.
 */
export function instanceOfInvoice(value) {
    let isInstance = true;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "amountDue" in value;
    isInstance = isInstance && "pdfUrl" in value;
    isInstance = isInstance && "dueDate" in value;
    isInstance = isInstance && "year" in value;
    isInstance = isInstance && "month" in value;
    isInstance = isInstance && "id" in value;
    return isInstance;
}
export function InvoiceFromJSON(json) {
    return InvoiceFromJSONTyped(json, false);
}
export function InvoiceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'status': json['status'],
        'amountDue': json['amountDue'],
        'pdfUrl': json['pdfUrl'],
        'dueDate': (new Date(json['dueDate'])),
        'year': json['year'],
        'month': json['month'],
        'id': json['id'],
    };
}
export function InvoiceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'status': value.status,
        'amountDue': value.amountDue,
        'pdfUrl': value.pdfUrl,
        'dueDate': (value.dueDate.toISOString()),
        'year': value.year,
        'month': value.month,
        'id': value.id,
    };
}
