// @hathora/hathora-cloud-sdk/dist/esm/models/LinkPaymentMethod.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the LinkPaymentMethod interface.
 */
export function instanceOfLinkPaymentMethod(value) {
    let isInstance = true;
    return isInstance;
}
export function LinkPaymentMethodFromJSON(json) {
    return LinkPaymentMethodFromJSONTyped(json, false);
}
export function LinkPaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'email': !exists(json, 'email') ? undefined : json['email'],
    };
}
export function LinkPaymentMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
    };
}
