// @hathora/hathora-cloud-sdk/dist/esm/models/LobbyVisibility.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Types of lobbies a player can create.
 *
 * `private`: the player who created the room must share the roomId with their friends
 *
 * `public`: visible in the public lobby list, anyone can join
 *
 * `local`: for testing with a server running locally
 * @export
 */
export const LobbyVisibility = {
    Private: 'private',
    Public: 'public',
    Local: 'local'
};
export function LobbyVisibilityFromJSON(json) {
    return LobbyVisibilityFromJSONTyped(json, false);
}
export function LobbyVisibilityFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
export function LobbyVisibilityToJSON(value) {
    return value;
}
