// @hathora/hathora-cloud-sdk/dist/esm/models/PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { AchPaymentMethodFromJSON, AchPaymentMethodToJSON, } from './AchPaymentMethod';
import { CardPaymentMethodFromJSON, CardPaymentMethodToJSON, } from './CardPaymentMethod';
import { LinkPaymentMethodFromJSON, LinkPaymentMethodToJSON, } from './LinkPaymentMethod';
/**
 * Check if a given object implements the PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod interface.
 */
export function instanceOfPartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod(value) {
    let isInstance = true;
    return isInstance;
}
export function PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSON(json) {
    return PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSONTyped(json, false);
}
export function PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'card': !exists(json, 'card') ? undefined : CardPaymentMethodFromJSON(json['card']),
        'ach': !exists(json, 'ach') ? undefined : AchPaymentMethodFromJSON(json['ach']),
        'link': !exists(json, 'link') ? undefined : LinkPaymentMethodFromJSON(json['link']),
    };
}
export function PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'card': CardPaymentMethodToJSON(value.card),
        'ach': AchPaymentMethodToJSON(value.ach),
        'link': LinkPaymentMethodToJSON(value.link),
    };
}
