// @hathora/hathora-cloud-sdk/dist/esm/models/RoomStatus.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * The allocation status of a room.
 *
 * `scheduling`: a process is not allocated yet and the room is waiting to be scheduled
 *
 * `active`: ready to accept connections
 *
 * `suspended`: room is unallocated from the process but can be rescheduled later with the same `roomId`
 *
 * `destroyed`: all associated metadata is deleted
 * @export
 */
export const RoomStatus = {
    Scheduling: 'scheduling',
    Active: 'active',
    Suspended: 'suspended',
    Destroyed: 'destroyed'
};
export function RoomStatusFromJSON(json) {
    return RoomStatusFromJSONTyped(json, false);
}
export function RoomStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
export function RoomStatusToJSON(value) {
    return value;
}
