// @hathora/hathora-cloud-sdk/dist/esm/models/VerificationEmailResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * @export
 */
export const VerificationEmailResponseStatusEnum = {
    Success: 'success'
};
/**
 * Check if a given object implements the VerificationEmailResponse interface.
 */
export function instanceOfVerificationEmailResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "status" in value;
    return isInstance;
}
export function VerificationEmailResponseFromJSON(json) {
    return VerificationEmailResponseFromJSONTyped(json, false);
}
export function VerificationEmailResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'status': json['status'],
    };
}
export function VerificationEmailResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'status': value.status,
    };
}
