// @hathora/hathora-cloud-sdk/dist/esm/models/VerificationEmailRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the VerificationEmailRequest interface.
 */
export function instanceOfVerificationEmailRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "userId" in value;
    return isInstance;
}
export function VerificationEmailRequestFromJSON(json) {
    return VerificationEmailRequestFromJSONTyped(json, false);
}
export function VerificationEmailRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'userId': json['userId'],
    };
}
export function VerificationEmailRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'userId': value.userId,
    };
}
