// @hathora/hathora-cloud-sdk/dist/esm/apis/ManagementV1Api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import * as runtime from '../runtime';
import { VerificationEmailRequestToJSON, VerificationEmailResponseFromJSON, } from '../models';
/**
 *
 */
export class ManagementV1Api extends runtime.BaseAPI {
    /**
     */
    sendVerificationEmailRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters.verificationEmailRequest === null || requestParameters.verificationEmailRequest === undefined) {
                throw new runtime.RequiredError('verificationEmailRequest', 'Required parameter requestParameters.verificationEmailRequest was null or undefined when calling sendVerificationEmail.');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            const response = yield this.request({
                path: `/management/v1/sendverificationemail`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: VerificationEmailRequestToJSON(requestParameters.verificationEmailRequest),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => VerificationEmailResponseFromJSON(jsonValue));
        });
    }
    /**
     */
    sendVerificationEmail(verificationEmailRequest, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.sendVerificationEmailRaw({ verificationEmailRequest: verificationEmailRequest }, initOverrides);
            return yield response.value();
        });
    }
}
