// @hathora/hathora-cloud-sdk/dist/models/AchPaymentMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchPaymentMethodToJSON = exports.AchPaymentMethodFromJSONTyped = exports.AchPaymentMethodFromJSON = exports.instanceOfAchPaymentMethod = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AchPaymentMethod interface.
 */
function instanceOfAchPaymentMethod(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfAchPaymentMethod = instanceOfAchPaymentMethod;
function AchPaymentMethodFromJSON(json) {
    return AchPaymentMethodFromJSONTyped(json, false);
}
exports.AchPaymentMethodFromJSON = AchPaymentMethodFromJSON;
function AchPaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'last4': !(0, runtime_1.exists)(json, 'last4') ? undefined : json['last4'],
        'bankName': !(0, runtime_1.exists)(json, 'bankName') ? undefined : json['bankName'],
    };
}
exports.AchPaymentMethodFromJSONTyped = AchPaymentMethodFromJSONTyped;
function AchPaymentMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'last4': value.last4,
        'bankName': value.bankName,
    };
}
exports.AchPaymentMethodToJSON = AchPaymentMethodToJSON;
