// @hathora/hathora-cloud-sdk/dist/models/AppConfigAuthConfiguration.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppConfigAuthConfigurationToJSON = exports.AppConfigAuthConfigurationFromJSONTyped = exports.AppConfigAuthConfigurationFromJSON = exports.instanceOfAppConfigAuthConfiguration = void 0;
const runtime_1 = require("../runtime");
const ApplicationAuthConfigurationGoogle_1 = require("./ApplicationAuthConfigurationGoogle");
/**
 * Check if a given object implements the AppConfigAuthConfiguration interface.
 */
function instanceOfAppConfigAuthConfiguration(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfAppConfigAuthConfiguration = instanceOfAppConfigAuthConfiguration;
function AppConfigAuthConfigurationFromJSON(json) {
    return AppConfigAuthConfigurationFromJSONTyped(json, false);
}
exports.AppConfigAuthConfigurationFromJSON = AppConfigAuthConfigurationFromJSON;
function AppConfigAuthConfigurationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !(0, runtime_1.exists)(json, 'google') ? undefined : (0, ApplicationAuthConfigurationGoogle_1.ApplicationAuthConfigurationGoogleFromJSON)(json['google']),
        'nickname': !(0, runtime_1.exists)(json, 'nickname') ? undefined : json['nickname'],
        'anonymous': !(0, runtime_1.exists)(json, 'anonymous') ? undefined : json['anonymous'],
    };
}
exports.AppConfigAuthConfigurationFromJSONTyped = AppConfigAuthConfigurationFromJSONTyped;
function AppConfigAuthConfigurationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': (0, ApplicationAuthConfigurationGoogle_1.ApplicationAuthConfigurationGoogleToJSON)(value.google),
        'nickname': value.nickname,
        'anonymous': value.anonymous,
    };
}
exports.AppConfigAuthConfigurationToJSON = AppConfigAuthConfigurationToJSON;
