// @hathora/hathora-cloud-sdk/dist/models/CardBrand.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardBrandToJSON = exports.CardBrandFromJSONTyped = exports.CardBrandFromJSON = exports.CardBrand = void 0;
/**
 *
 * @export
 */
exports.CardBrand = {
    Amex: 'amex',
    CartesBancaires: 'cartes_bancaires',
    Diners: 'diners',
    Discover: 'discover',
    Jcb: 'jcb',
    Mastercard: 'mastercard',
    Visa: 'visa',
    Unionpay: 'unionpay',
    Card: 'card'
};
function CardBrandFromJSON(json) {
    return CardBrandFromJSONTyped(json, false);
}
exports.CardBrandFromJSON = CardBrandFromJSON;
function CardBrandFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.CardBrandFromJSONTyped = CardBrandFromJSONTyped;
function CardBrandToJSON(value) {
    return value;
}
exports.CardBrandToJSON = CardBrandToJSON;
