// @hathora/hathora-cloud-sdk/dist/models/CardPaymentMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardPaymentMethodToJSON = exports.CardPaymentMethodFromJSONTyped = exports.CardPaymentMethodFromJSON = exports.instanceOfCardPaymentMethod = void 0;
const CardBrand_1 = require("./CardBrand");
/**
 * Check if a given object implements the CardPaymentMethod interface.
 */
function instanceOfCardPaymentMethod(value) {
    let isInstance = true;
    isInstance = isInstance && "last4" in value;
    isInstance = isInstance && "brand" in value;
    return isInstance;
}
exports.instanceOfCardPaymentMethod = instanceOfCardPaymentMethod;
function CardPaymentMethodFromJSON(json) {
    return CardPaymentMethodFromJSONTyped(json, false);
}
exports.CardPaymentMethodFromJSON = CardPaymentMethodFromJSON;
function CardPaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'last4': json['last4'],
        'brand': (0, CardBrand_1.CardBrandFromJSON)(json['brand']),
    };
}
exports.CardPaymentMethodFromJSONTyped = CardPaymentMethodFromJSONTyped;
function CardPaymentMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'last4': value.last4,
        'brand': (0, CardBrand_1.CardBrandToJSON)(value.brand),
    };
}
exports.CardPaymentMethodToJSON = CardPaymentMethodToJSON;
