// @hathora/hathora-cloud-sdk/dist/models/ContainerPort.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerPortToJSON = exports.ContainerPortFromJSONTyped = exports.ContainerPortFromJSON = exports.instanceOfContainerPort = void 0;
const TransportType_1 = require("./TransportType");
/**
 * Check if a given object implements the ContainerPort interface.
 */
function instanceOfContainerPort(value) {
    let isInstance = true;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfContainerPort = instanceOfContainerPort;
function ContainerPortFromJSON(json) {
    return ContainerPortFromJSONTyped(json, false);
}
exports.ContainerPortFromJSON = ContainerPortFromJSON;
function ContainerPortFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'transportType': (0, TransportType_1.TransportTypeFromJSON)(json['transportType']),
        'port': json['port'],
        'name': json['name'],
    };
}
exports.ContainerPortFromJSONTyped = ContainerPortFromJSONTyped;
function ContainerPortToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transportType': (0, TransportType_1.TransportTypeToJSON)(value.transportType),
        'port': value.port,
        'name': value.name,
    };
}
exports.ContainerPortToJSON = ContainerPortToJSON;
