// @hathora/hathora-cloud-sdk/dist/models/CreateRoomDeprecatedRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRoomDeprecatedRequestToJSON = exports.CreateRoomDeprecatedRequestFromJSONTyped = exports.CreateRoomDeprecatedRequestFromJSON = exports.instanceOfCreateRoomDeprecatedRequest = void 0;
const Region_1 = require("./Region");
/**
 * Check if a given object implements the CreateRoomDeprecatedRequest interface.
 */
function instanceOfCreateRoomDeprecatedRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
exports.instanceOfCreateRoomDeprecatedRequest = instanceOfCreateRoomDeprecatedRequest;
function CreateRoomDeprecatedRequestFromJSON(json) {
    return CreateRoomDeprecatedRequestFromJSONTyped(json, false);
}
exports.CreateRoomDeprecatedRequestFromJSON = CreateRoomDeprecatedRequestFromJSON;
function CreateRoomDeprecatedRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'region': (0, Region_1.RegionFromJSON)(json['region']),
    };
}
exports.CreateRoomDeprecatedRequestFromJSONTyped = CreateRoomDeprecatedRequestFromJSONTyped;
function CreateRoomDeprecatedRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'region': (0, Region_1.RegionToJSON)(value.region),
    };
}
exports.CreateRoomDeprecatedRequestToJSON = CreateRoomDeprecatedRequestToJSON;
