// @hathora/hathora-cloud-sdk/dist/models/CreateLobbyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateLobbyRequestToJSON = exports.CreateLobbyRequestFromJSONTyped = exports.CreateLobbyRequestFromJSON = exports.instanceOfCreateLobbyRequest = void 0;
const LobbyVisibility_1 = require("./LobbyVisibility");
const Region_1 = require("./Region");
/**
 * Check if a given object implements the CreateLobbyRequest interface.
 */
function instanceOfCreateLobbyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "visibility" in value;
    isInstance = isInstance && "initialConfig" in value;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
exports.instanceOfCreateLobbyRequest = instanceOfCreateLobbyRequest;
function CreateLobbyRequestFromJSON(json) {
    return CreateLobbyRequestFromJSONTyped(json, false);
}
exports.CreateLobbyRequestFromJSON = CreateLobbyRequestFromJSON;
function CreateLobbyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'visibility': (0, LobbyVisibility_1.LobbyVisibilityFromJSON)(json['visibility']),
        'initialConfig': json['initialConfig'],
        'region': (0, Region_1.RegionFromJSON)(json['region']),
    };
}
exports.CreateLobbyRequestFromJSONTyped = CreateLobbyRequestFromJSONTyped;
function CreateLobbyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'visibility': (0, LobbyVisibility_1.LobbyVisibilityToJSON)(value.visibility),
        'initialConfig': value.initialConfig,
        'region': (0, Region_1.RegionToJSON)(value.region),
    };
}
exports.CreateLobbyRequestToJSON = CreateLobbyRequestToJSON;
