// @hathora/hathora-cloud-sdk/dist/models/CustomerPortalUrl.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalUrlToJSON = exports.CustomerPortalUrlFromJSONTyped = exports.CustomerPortalUrlFromJSON = exports.instanceOfCustomerPortalUrl = void 0;
/**
 * Check if a given object implements the CustomerPortalUrl interface.
 */
function instanceOfCustomerPortalUrl(value) {
    let isInstance = true;
    isInstance = isInstance && "returnUrl" in value;
    return isInstance;
}
exports.instanceOfCustomerPortalUrl = instanceOfCustomerPortalUrl;
function CustomerPortalUrlFromJSON(json) {
    return CustomerPortalUrlFromJSONTyped(json, false);
}
exports.CustomerPortalUrlFromJSON = CustomerPortalUrlFromJSON;
function CustomerPortalUrlFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'returnUrl': json['returnUrl'],
    };
}
exports.CustomerPortalUrlFromJSONTyped = CustomerPortalUrlFromJSONTyped;
function CustomerPortalUrlToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'returnUrl': value.returnUrl,
    };
}
exports.CustomerPortalUrlToJSON = CustomerPortalUrlToJSON;
