// @hathora/hathora-cloud-sdk/dist/models/DeploymentEnvInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentEnvInnerToJSON = exports.DeploymentEnvInnerFromJSONTyped = exports.DeploymentEnvInnerFromJSON = exports.instanceOfDeploymentEnvInner = void 0;
/**
 * Check if a given object implements the DeploymentEnvInner interface.
 */
function instanceOfDeploymentEnvInner(value) {
    let isInstance = true;
    isInstance = isInstance && "value" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfDeploymentEnvInner = instanceOfDeploymentEnvInner;
function DeploymentEnvInnerFromJSON(json) {
    return DeploymentEnvInnerFromJSONTyped(json, false);
}
exports.DeploymentEnvInnerFromJSON = DeploymentEnvInnerFromJSON;
function DeploymentEnvInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': json['value'],
        'name': json['name'],
    };
}
exports.DeploymentEnvInnerFromJSONTyped = DeploymentEnvInnerFromJSONTyped;
function DeploymentEnvInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': value.value,
        'name': value.name,
    };
}
exports.DeploymentEnvInnerToJSON = DeploymentEnvInnerToJSON;
