// @hathora/hathora-cloud-sdk/dist/models/DeploymentConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentConfigToJSON = exports.DeploymentConfigFromJSONTyped = exports.DeploymentConfigFromJSON = exports.instanceOfDeploymentConfig = void 0;
const runtime_1 = require("../runtime");
const ContainerPort_1 = require("./ContainerPort");
const DeploymentEnvInner_1 = require("./DeploymentEnvInner");
const PlanName_1 = require("./PlanName");
const TransportType_1 = require("./TransportType");
/**
 * Check if a given object implements the DeploymentConfig interface.
 */
function instanceOfDeploymentConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "env" in value;
    isInstance = isInstance && "roomsPerProcess" in value;
    isInstance = isInstance && "planName" in value;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "containerPort" in value;
    return isInstance;
}
exports.instanceOfDeploymentConfig = instanceOfDeploymentConfig;
function DeploymentConfigFromJSON(json) {
    return DeploymentConfigFromJSONTyped(json, false);
}
exports.DeploymentConfigFromJSON = DeploymentConfigFromJSON;
function DeploymentConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'env': (json['env'].map(DeploymentEnvInner_1.DeploymentEnvInnerFromJSON)),
        'roomsPerProcess': json['roomsPerProcess'],
        'planName': (0, PlanName_1.PlanNameFromJSON)(json['planName']),
        'additionalContainerPorts': !(0, runtime_1.exists)(json, 'additionalContainerPorts') ? undefined : (json['additionalContainerPorts'].map(ContainerPort_1.ContainerPortFromJSON)),
        'transportType': (0, TransportType_1.TransportTypeFromJSON)(json['transportType']),
        'containerPort': json['containerPort'],
    };
}
exports.DeploymentConfigFromJSONTyped = DeploymentConfigFromJSONTyped;
function DeploymentConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'env': (value.env.map(DeploymentEnvInner_1.DeploymentEnvInnerToJSON)),
        'roomsPerProcess': value.roomsPerProcess,
        'planName': (0, PlanName_1.PlanNameToJSON)(value.planName),
        'additionalContainerPorts': value.additionalContainerPorts === undefined ? undefined : (value.additionalContainerPorts.map(ContainerPort_1.ContainerPortToJSON)),
        'transportType': (0, TransportType_1.TransportTypeToJSON)(value.transportType),
        'containerPort': value.containerPort,
    };
}
exports.DeploymentConfigToJSON = DeploymentConfigToJSON;
