// @hathora/hathora-cloud-sdk/dist/models/ExposedPort.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExposedPortToJSON = exports.ExposedPortFromJSONTyped = exports.ExposedPortFromJSON = exports.instanceOfExposedPort = void 0;
const TransportType_1 = require("./TransportType");
/**
 * Check if a given object implements the ExposedPort interface.
 */
function instanceOfExposedPort(value) {
    let isInstance = true;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "host" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfExposedPort = instanceOfExposedPort;
function ExposedPortFromJSON(json) {
    return ExposedPortFromJSONTyped(json, false);
}
exports.ExposedPortFromJSON = ExposedPortFromJSON;
function ExposedPortFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'transportType': (0, TransportType_1.TransportTypeFromJSON)(json['transportType']),
        'port': json['port'],
        'host': json['host'],
        'name': json['name'],
    };
}
exports.ExposedPortFromJSONTyped = ExposedPortFromJSONTyped;
function ExposedPortToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transportType': (0, TransportType_1.TransportTypeToJSON)(value.transportType),
        'port': value.port,
        'host': value.host,
        'name': value.name,
    };
}
exports.ExposedPortToJSON = ExposedPortToJSON;
