// @hathora/hathora-cloud-sdk/dist/models/LinkPaymentMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkPaymentMethodToJSON = exports.LinkPaymentMethodFromJSONTyped = exports.LinkPaymentMethodFromJSON = exports.instanceOfLinkPaymentMethod = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the LinkPaymentMethod interface.
 */
function instanceOfLinkPaymentMethod(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfLinkPaymentMethod = instanceOfLinkPaymentMethod;
function LinkPaymentMethodFromJSON(json) {
    return LinkPaymentMethodFromJSONTyped(json, false);
}
exports.LinkPaymentMethodFromJSON = LinkPaymentMethodFromJSON;
function LinkPaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
    };
}
exports.LinkPaymentMethodFromJSONTyped = LinkPaymentMethodFromJSONTyped;
function LinkPaymentMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
    };
}
exports.LinkPaymentMethodToJSON = LinkPaymentMethodToJSON;
