// @hathora/hathora-cloud-sdk/dist/models/LobbyVisibility.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LobbyVisibilityToJSON = exports.LobbyVisibilityFromJSONTyped = exports.LobbyVisibilityFromJSON = exports.LobbyVisibility = void 0;
/**
 * Types of lobbies a player can create.
 *
 * `private`: the player who created the room must share the roomId with their friends
 *
 * `public`: visible in the public lobby list, anyone can join
 *
 * `local`: for testing with a server running locally
 * @export
 */
exports.LobbyVisibility = {
    Private: 'private',
    Public: 'public',
    Local: 'local'
};
function LobbyVisibilityFromJSON(json) {
    return LobbyVisibilityFromJSONTyped(json, false);
}
exports.LobbyVisibilityFromJSON = LobbyVisibilityFromJSON;
function LobbyVisibilityFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.LobbyVisibilityFromJSONTyped = LobbyVisibilityFromJSONTyped;
function LobbyVisibilityToJSON(value) {
    return value;
}
exports.LobbyVisibilityToJSON = LobbyVisibilityToJSON;
