// @hathora/hathora-cloud-sdk/dist/models/LoginResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginResponseToJSON = exports.LoginResponseFromJSONTyped = exports.LoginResponseFromJSON = exports.instanceOfLoginResponse = void 0;
/**
 * Check if a given object implements the LoginResponse interface.
 */
function instanceOfLoginResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
exports.instanceOfLoginResponse = instanceOfLoginResponse;
function LoginResponseFromJSON(json) {
    return LoginResponseFromJSONTyped(json, false);
}
exports.LoginResponseFromJSON = LoginResponseFromJSON;
function LoginResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'token': json['token'],
    };
}
exports.LoginResponseFromJSONTyped = LoginResponseFromJSONTyped;
function LoginResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'token': value.token,
    };
}
exports.LoginResponseToJSON = LoginResponseToJSON;
