// @hathora/hathora-cloud-sdk/dist/models/PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodToJSON = exports.PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSONTyped = exports.PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSON = exports.instanceOfPartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod = void 0;
const runtime_1 = require("../runtime");
const AchPaymentMethod_1 = require("./AchPaymentMethod");
const CardPaymentMethod_1 = require("./CardPaymentMethod");
const LinkPaymentMethod_1 = require("./LinkPaymentMethod");
/**
 * Check if a given object implements the PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod interface.
 */
function instanceOfPartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod = instanceOfPartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethod;
function PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSON(json) {
    return PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSONTyped(json, false);
}
exports.PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSON = PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSON;
function PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'card': !(0, runtime_1.exists)(json, 'card') ? undefined : (0, CardPaymentMethod_1.CardPaymentMethodFromJSON)(json['card']),
        'ach': !(0, runtime_1.exists)(json, 'ach') ? undefined : (0, AchPaymentMethod_1.AchPaymentMethodFromJSON)(json['ach']),
        'link': !(0, runtime_1.exists)(json, 'link') ? undefined : (0, LinkPaymentMethod_1.LinkPaymentMethodFromJSON)(json['link']),
    };
}
exports.PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSONTyped = PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSONTyped;
function PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'card': (0, CardPaymentMethod_1.CardPaymentMethodToJSON)(value.card),
        'ach': (0, AchPaymentMethod_1.AchPaymentMethodToJSON)(value.ach),
        'link': (0, LinkPaymentMethod_1.LinkPaymentMethodToJSON)(value.link),
    };
}
exports.PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodToJSON = PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodToJSON;
