// @hathora/hathora-cloud-sdk/dist/models/RoomStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoomStatusToJSON = exports.RoomStatusFromJSONTyped = exports.RoomStatusFromJSON = exports.RoomStatus = void 0;
/**
 * The allocation status of a room.
 *
 * `scheduling`: a process is not allocated yet and the room is waiting to be scheduled
 *
 * `active`: ready to accept connections
 *
 * `suspended`: room is unallocated from the process but can be rescheduled later with the same `roomId`
 *
 * `destroyed`: all associated metadata is deleted
 * @export
 */
exports.RoomStatus = {
    Scheduling: 'scheduling',
    Active: 'active',
    Suspended: 'suspended',
    Destroyed: 'destroyed'
};
function RoomStatusFromJSON(json) {
    return RoomStatusFromJSONTyped(json, false);
}
exports.RoomStatusFromJSON = RoomStatusFromJSON;
function RoomStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.RoomStatusFromJSONTyped = RoomStatusFromJSONTyped;
function RoomStatusToJSON(value) {
    return value;
}
exports.RoomStatusToJSON = RoomStatusToJSON;
