// @hathora/hathora-cloud-sdk/dist/models/VerificationEmailRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerificationEmailRequestToJSON = exports.VerificationEmailRequestFromJSONTyped = exports.VerificationEmailRequestFromJSON = exports.instanceOfVerificationEmailRequest = void 0;
/**
 * Check if a given object implements the VerificationEmailRequest interface.
 */
function instanceOfVerificationEmailRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "userId" in value;
    return isInstance;
}
exports.instanceOfVerificationEmailRequest = instanceOfVerificationEmailRequest;
function VerificationEmailRequestFromJSON(json) {
    return VerificationEmailRequestFromJSONTyped(json, false);
}
exports.VerificationEmailRequestFromJSON = VerificationEmailRequestFromJSON;
function VerificationEmailRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'userId': json['userId'],
    };
}
exports.VerificationEmailRequestFromJSONTyped = VerificationEmailRequestFromJSONTyped;
function VerificationEmailRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'userId': value.userId,
    };
}
exports.VerificationEmailRequestToJSON = VerificationEmailRequestToJSON;
