// @hathora/hathora-cloud-sdk/dist/apis/BillingV1Api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * Welcome to the Hathora Cloud API documentation! Learn how to use the Hathora Cloud APIs to build and scale your game servers globally.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingV1Api = void 0;
const runtime = require("../runtime");
const models_1 = require("../models");
/**
 *
 */
class BillingV1Api extends runtime.BaseAPI {
    /**
     */
    getBalanceRaw(initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("auth0", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/billing/v1/balance`,
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            if (this.isJsonMime(response.headers.get('content-type'))) {
                return new runtime.JSONApiResponse(response);
            }
            else {
                return new runtime.TextApiResponse(response);
            }
        });
    }
    /**
     */
    getBalance(initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getBalanceRaw(initOverrides);
            return yield response.value();
        });
    }
    /**
     */
    getInvoicesRaw(initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("auth0", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/billing/v1/invoices`,
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(models_1.InvoiceFromJSON));
        });
    }
    /**
     */
    getInvoices(initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getInvoicesRaw(initOverrides);
            return yield response.value();
        });
    }
    /**
     */
    getPaymentMethodRaw(initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("auth0", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/billing/v1/paymentmethod`,
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => (0, models_1.PartialCardCardPaymentMethodAchAchPaymentMethodLinkLinkPaymentMethodFromJSON)(jsonValue));
        });
    }
    /**
     */
    getPaymentMethod(initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getPaymentMethodRaw(initOverrides);
            return yield response.value();
        });
    }
    /**
     */
    initStripeCustomerPortalUrlRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters.customerPortalUrl === null || requestParameters.customerPortalUrl === undefined) {
                throw new runtime.RequiredError('customerPortalUrl', 'Required parameter requestParameters.customerPortalUrl was null or undefined when calling initStripeCustomerPortalUrl.');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("auth0", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/billing/v1/customerportalurl`,
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: (0, models_1.CustomerPortalUrlToJSON)(requestParameters.customerPortalUrl),
            }, initOverrides);
            if (this.isJsonMime(response.headers.get('content-type'))) {
                return new runtime.JSONApiResponse(response);
            }
            else {
                return new runtime.TextApiResponse(response);
            }
        });
    }
    /**
     */
    initStripeCustomerPortalUrl(customerPortalUrl, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.initStripeCustomerPortalUrlRaw({ customerPortalUrl: customerPortalUrl }, initOverrides);
            return yield response.value();
        });
    }
}
exports.BillingV1Api = BillingV1Api;
